### PR TITLE
add darwin tun support

### PR DESCRIPTION
**Pull request by wsvn53**
Support open tun device on Mac OS X. Tap device not support yet.

### CODE REVIEW

File: taptun_darwin.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/*
```
Comment (davecheney) [inline taptun_darwin.h]: please remove this boilerplate

File: taptun_darwin.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/*
```
Comment (davecheney) [inline taptun_darwin.c]: please remove this boilerplate

**Comment by davecheney** [comment]
Thank you for this submission. Please remove the boilerplate and I'll merge this, thanks.

**Comment by wsvn53** [comment]
done! boilerplate removed.

**Comment by davecheney** [comment]
Thank you.

**Pull request merged**